inc/Engine/Optimization/RUCSS/Admin/Subscriber.php

```diff
@@ -136,6 +139,7 @@ public function truncate_used_css() {
 		}
 
 		$this->database->truncate_used_css_table();
+		$this->used_css->delete_all_used_css_files();
```

Comment:
I don't think we need this in here.

For theme change, indeed the full WP Rocket cache will be cleaned. However this truncate_used_css() will be triggered also on: `rocket_rucss_file_changed` hook which do not need to clean also the files.

Reply:
When WP Rocket cache is cleaned, `used-css` directory isn't removed.

But I think really we need to remove it from here but not sure.

Reply:
It should NOT be removed from here. Why? This function is hooked in 2 places: 
```
'switch_theme'                       => 'truncate_used_css',
'rocket_rucss_file_changed'          => 'truncate_used_css',
```

On switch_theme we clean the WP Rocket cache, right? Then we can safely delete also the used_css directory.
However, when is `rocket_rucss_file_changed` triggererd? From Warmup side, in case we identify a used css that was changed. During that operation we don't clean also WP Rocket cache, so we will end up in the 404 situation.

So, cleaning the used-css folder should happen only if the page cache file is deleted. Otherwise: 404 and broken page style.

Reply:
Theoretically, `rocket_rucss_file_changed` is called just after sending all resources to the warmup before calling treeshaker on the cache file generation and before putting the used_css file into the cached page.

But to be on the safe side, I will remove it only with `switch_theme`.

Reply:
Done

Reply:
And `rocket_rucss_file_changed` could be called for a single page where we identify a new CSS file which do not exist on other pages. This means, that the cache will be created only for that specific page and the full cache will not be regenerating.

So cleaning the full used css file when we identify a single resource changed does not make sense.

What do you think?

Reply:
Exactly.

You can check it now, I adjusted it.